### PR TITLE
Rewriting UsersCreation.cy.ts according to POM approach

### LINKED ISSUE #9255: Refactor Phone number generation logic in cypress to handle security concern raised by codeQL

### PR DESCRIPTION
**Pull request by AnveshNalimela**
## Proposed Changes

- Fixes #https://github.com/ohcnetwork/care_fe/issues/8929
- replace all the clickElementById() custom function usage in the cypress test file with individual proper functions
- use all reusable components from commands.ts wherever applicable

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced new methods for user profile management, enhancing the clarity of user interactions.
	- Added methods for clearing input fields, improving user experience during data entry.

- **Bug Fixes**
	- Renamed methods for better clarity, ensuring users can easily understand their functionality.

- **Refactor**
	- Improved organization and readability of user creation and profile update tests.
	- Enhanced method specificity and clarity in user creation and profile management.
	- Updated functions for generating phone numbers to improve randomness and security.

- **Documentation**
	- Updated method names and parameters to follow consistent naming conventions for better usability.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -15,3 +15,4 @@ CARE_CDN_URL="https://egov-s3-facility-10bedicu.s3.amazonaws.com https://egov-s3
 REACT_ALLOWED_LOCALES="en,hi,ta,ml,mr,kn"
 
 REACT_ENABLED_APPS="ohcnetwork/care_livekit_fe@main"
+REACT_CARE_API_URL=http://127.0.0.1:9000
```
Comment (shivankacker) [inline .env]:
```suggestion
```
Refrain from updating `.env` files. Create a `.env.local` file for development

File: cypress/e2e/users_spec/UsersCreation.cy.ts
Comment (shivankacker) [inline cypress/e2e/users_spec/UsersCreation.cy.ts:1]: @AnveshNalimela Remove the declaration of the clickelementbyid function as well. There are more usages of the function in `cypress/pageobject/Users/UserCreation.ts`

**Review by shivankacker — CHANGES_REQUESTED** [review]

File: cypress/e2e/users_spec/UsersCreation.cy.ts
Comment (AnveshNalimela) [reply cypress/e2e/users_spec/UsersCreation.cy.ts:1]: Made the required changes! Have a look
@shivankacker

File: .env [hunk 2]
```diff
@@ -15,3 +15,4 @@ CARE_CDN_URL="https://egov-s3-facility-10bedicu.s3.amazonaws.com https://egov-s3
 REACT_ALLOWED_LOCALES="en,hi,ta,ml,mr,kn"
 
 REACT_ENABLED_APPS="ohcnetwork/care_livekit_fe@main"
+
```
Comment (shivankacker) [inline .env]:
```suggestion
```
No unwanted changes should be pushed, even blank lines

**Review by shivankacker — APPROVED** [review]

**Review by nihal467 — CHANGES_REQUESTED** [review]
1. We use POM approach in cypress, so changes everything into a re-usable function Refer: PatientDoctorConnect.cy.ts test file

For example cy.get("#user-profile-name").click(); should be a function which is named clickUserProfileName()

Similarly, convert the entire test file to the POM approach

**Comment by AnveshNalimela** [comment]
@nihal467 
have a look at this 
```
clickProfileName() {
    cy.get("#user-profile-name").click();
  }
```
Declare such indiviual functions in pageobejct of user creation.ts

**Comment by nihal467** [comment]
@AnveshNalimela i have modified the issue, can you clear the rest of them ?

**Comment by AnveshNalimela** [comment]
@nihal467  made the changes accordingly by refering patientDoctorConnect.cy.ts .just have look at latest commit

**Comment by nihal467** [comment]
@AnveshNalimela as per the updated issue , you need to rewrite the entire file, read it carefully, so which means that, we need to give individual function name to each operation.


image

Replace all the usage of : 

- Removed the clickelemenenbyid with individual function name following the POM
- similar do the above process for clearintoelementbyid, typeintoelemenetbyid
- replace the setinputdate , selectdropdownoption, verifynotification, verifyelementcontainstext,verifyerrormessage with already setup command.ts function

inshort: rewrite the entire file with new individual proper function, proper usage of command.ts functions, and keep it clean

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 3]
```diff
@@ -66,13 +66,13 @@ describe("User Creation", () => {
   });
 
   it("Update the existing user profile and verify its reflection", () => {
-    userCreationPage.clickElementById("user-profile-name");
-    userCreationPage.clickElementById("profile-button");
+    userCreationPage.clickProfileName();
+    userCreationPage.clickProfileButton();
     userCreationPage.verifyElementContainsText(
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]: use the command.ts reusable function here for verifying elements presence

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 4]
```diff
@@ -66,13 +66,13 @@ describe("User Creation", () => {
...
       "username-profile-details",
       "devdistrictadmin",
     );
-    userCreationPage.clickElementById("edit-cancel-profile-button");
+    userCreationPage.clickEditCancelProfileButton();
     userCreationPage.typeIntoElementByIdPostClear(
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]: remove all usage of typeintoelementbyidpostclear all those function with proper individual POM approached function

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 5]
```diff
@@ -90,7 +90,7 @@ describe("User Creation", () => {
       "date_of_birth",
       "01011998",
     );
-    userCreationPage.clickElementById("submit");
+    cy.submitButton("Update");
     userCreationPage.verifyElementContainsText(
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]: usage the command.ts reusable function for verifying content presence

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 6]
```diff
@@ -126,20 +126,21 @@ describe("User Creation", () => {
...
 
   it("Update the existing user profile Form Mandatory File Error", () => {
-    userCreationPage.clickElementById("user-profile-name");
-    userCreationPage.clickElementById("profile-button");
-    userCreationPage.clickElementById("edit-cancel-profile-button");
+    userCreationPage.clickProfileName();
+    userCreationPage.clickProfileButton();
+    userCreationPage.clickEditCancelProfileButton();
+
     userCreationPage.clearIntoElementById("firstName");
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]: remove the usage of clearintoelementbyid and use individual proper function for each action with POM approach

**Comment by AnveshNalimela** [comment]
I will be on it

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 7]
```diff
@@ -126,20 +126,21 @@ describe("User Creation", () => {
...
+    cy.submitButton("Update");
     userCreationPage.verifyErrorMessages(EXPECTED_PROFILE_ERROR_MESSAGES);
   });
 
   it("create new user and verify reflection", () => {
-    userCreationPage.clickElementById("addUserButton");
+    userCreationPage.clickAddUserButton();
     userCreationPage.selectFacility("Dummy Shifting Center");
     userCreationPage.typeIntoElementById("username", username);
     userCreationPage.typeIntoElementById("password", "Test@123");
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]: remove all usage of typeintoelementbyId and create POM approach based individual functions

File: cypress/pageobject/Users/UserCreation.ts [hunk 8]
```diff
@@ -59,12 +74,12 @@ export class UserCreationPage {
     inputElementId: string,
     dateValue: string,
   ) {
-    this.clickElementById(dateElementId);
+    cy.get("#" + dateElementId).click();
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts]: we have a reusable command.ts function, for date selection component , refer it

File: cypress/pageobject/Users/UserCreation.ts [hunk 9]
```diff
@@ -59,12 +74,12 @@ export class UserCreationPage {
...
     this.typeIntoElementById(inputElementId, dateValue);
   }
 
   selectDropdownOption(dropdownId: string, optionText: string) {
-    this.clickElementById(dropdownId);
+    cy.get("#" + dropdownId).click();
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts]: we have reusable command.ts function, for all type of dropdown , refer it

**Comment by AnveshNalimela** [comment]
@nihal467  Made the changes. Have a look

File: cypress/e2e/users_spec/UsersCreation.cy.ts (hunk 7 above)
Comment (AnveshNalimela) [reply cypress/e2e/users_spec/UsersCreation.cy.ts]: Done

File: cypress/pageobject/Users/UserCreation.ts (hunk 8 above)
Comment (AnveshNalimela) [reply cypress/pageobject/Users/UserCreation.ts]: It is just a click. I dont think we have function for simple click in data selection

**Comment by AnveshNalimela** [comment]
@nihal467  Looking forward for response ..

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 10]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
-    userCreationPage.typeIntoElementByIdPostClear(
-      "altPhoneNumber",
-      emergency_phone_number,
-    );
-    userCreationPage.typeIntoElementByIdPostClear("email", "test@test.com");
-    userCreationPage.typeIntoElementByIdPostClear("weekly_working_hours", "14");
+    userCreationPage.clickProfileName();
+    userCreationPage.clickProfileButton();
+    cy.verifyContentPresence("#username-profile-details", ["devdistrictadmin"]);
+
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]: remove this unwanted space

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 11]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    userCreationPage.clickEditCancelProfileButton();
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    userCreationPage.clickEditProfileButton();
```

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 12]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    userCreationPage.typeIntoFirstNameAndClear();
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    userCreationPage.typeFirstName(firstName);
```

- separate the clear operation from this function and create a separate function for clearing functionality
- declare a const firstName = "District Editted" and use it here
- always keep the inputting variable in the test file not in the page object file

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 13]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    userCreationPage.typeIntoLastNameAndClear();
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    userCreationPage.typeLastName(lastName);
```

- make all the similar changes i mention for firstName in this function as well

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 14]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    cy.clickAndSelectOption("#gender", "Male");
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    cy.clickAndSelectOption("#gender", "Male");
```

- create a new function selectGender(Gender);
- create a new const gender = "Male";

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 15]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    userCreationPage.typeIntoPhoneNumberAndClear(phone_number);
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    userCreationPage.typeIntoPhoneNumber(phone_number);
```

- separate the clear operation from this function and create a separate function for clearing functionality

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 16]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    userCreationPage.typeIntoAltPhoneumberAndClear(emergency_phone_number);
+    userCreationPage.typeIntoEmailAndClear();
+    userCreationPage.typeIntoWeeklyWorkingHoursAndClear();
     userCreationPage.typeIntoElementByIdPostClearDob(
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    userCreationPage.typeIntoElementByIdPostClearDob(
```

change this to use cy.clickAndTypeDate();

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 17]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
-    userCreationPage.verifyElementContainsText(
-      "gender-profile-details",
-      "Male",
-    );
-    userCreationPage.verifyElementContainsText(
-      "averageworkinghour-profile-details",
-      "14",
-    );
+    ]);
+    cy.verifyContentPresence("#lastname-profile-details", ["Cypress"]);
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    cy.verifyContentPresence("#lastname-profile-details", [lastName]);
```

- move all the input values to a const
- then use the const for input as well as data verification purpose

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 18]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    cy.verifyContentPresence("#date_of_birth-profile-details", ["01/01/1998"]);
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]: change similars as above comment
```suggestion
    cy.verifyContentPresence("#date_of_birth-profile-details", ["01/01/1998"]);
```

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 19]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    cy.verifyContentPresence("#emailid-profile-details", ["test@test.com"]);
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    cy.verifyContentPresence("#emailid-profile-details", ["test@test.com"]);
```

change similars as above comment

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 20]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    cy.verifyContentPresence("#gender-profile-details", ["Male"]);
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    cy.verifyContentPresence("#gender-profile-details", ["Male"]);
```

change similars as above comment

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 21]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    cy.verifyContentPresence("#averageworkinghour-profile-details", ["14"]);
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    cy.verifyContentPresence("#averageworkinghour-profile-details", ["14"]);
```

change similars as above comment

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 22]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
-    userCreationPage.typeIntoElementById("first_name", "cypress test");
-    userCreationPage.typeIntoElementById("last_name", "staff user");
-    userCreationPage.typeIntoElementById("email", "test@test.com");
-    userCreationPage.selectDropdownOption("gender", "Male");
-    userCreationPage.selectDropdownOption("state", "Kerala");
-    userCreationPage.selectDropdownOption("district", "Ernakulam");
-    userCreationPage.clickElementById("submit");
-    userCreationPage.verifyNotification("User added successfully");
+    userCreationPage.typePhoneNumber(phone_number);
+    cy.clickAndTypeDate("#date_of_birth", "25081999");
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    cy.clickAndTypeDate("#date_of_birth", "25081999");
```

- create a new function called typeUserDob and follow general guideline i already gave

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 23]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    cy.clickAndSelectOption("#user_type", "Doctor");
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    cy.clickAndSelectOption("#user_type", "Doctor");
```

- create a new function called selectUserType() and follow general guideline i already gave

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 24]
```diff
@@ -66,122 +66,87 @@ describe("User Creation", () => {
...
+    userCreationPage.typeQualification();
+    userCreationPage.typeDoctorExperience();
+    userCreationPage.typeDoctorMedicalCouncilRegNo();
+    userCreationPage.typeFirstName();
+    userCreationPage.typeLastName();
+    userCreationPage.typeEmail();
+    cy.clickAndSelectOption("#gender", "Male");
+    cy.clickAndSelectOption("#state", "Kerala");
+    cy.clickAndSelectOption("#district", "Ernakulam");
+    userCreationPage.clickSubmit();
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
    userCreationPage.clickSubmit();
```

use the cy.submitButton reusable component

**Review by nihal467 — CHANGES_REQUESTED** [review]
- in general, keep all the input value as const and use that for input and presence verification
- always separate input field clearing and inputting functionality
- keep the input values in the test file, not in the pageobject file
- create new function for things you already missed in your test
- keep the function name more appropriate towards the action

Refer test file like PatientLogUpdate.cy.ts if you have doubt, how the test should be written

**Comment by nihal467** [comment]
@AnveshNalimela make sure the GitHub actions are passing as well before requesting next review

File: cypress/e2e/users_spec/UsersCreation.cy.ts (hunk 10 above)
Comment (AnveshNalimela) [reply cypress/e2e/users_spec/UsersCreation.cy.ts]: https://github.com/AnveshNalimela/care_fe/blob/c4569fc16cb7f7f2d29552d660134c1868df9599/cypress/e2e/users_spec/UsersCreation.cy.ts#L93

File: cypress/e2e/users_spec/UsersCreation.cy.ts (hunk 11 above)
Comment (AnveshNalimela) [reply cypress/e2e/users_spec/UsersCreation.cy.ts]: https://github.com/AnveshNalimela/care_fe/blob/c4569fc16cb7f7f2d29552d660134c1868df9599/cypress/e2e/users_spec/UsersCreation.cy.ts#L94

File: cypress/e2e/users_spec/UsersCreation.cy.ts (hunk 13 above)
Comment (AnveshNalimela) [reply cypress/e2e/users_spec/UsersCreation.cy.ts]: {598225F2-79B7-4202-B3B1-48EEE28DFACE}

File: cypress/e2e/users_spec/UsersCreation.cy.ts (hunk 20 above)
Comment (AnveshNalimela) [reply cypress/e2e/users_spec/UsersCreation.cy.ts]: {595A5EF1-60A7-40B5-B0E0-36AE2F5C5822}

File: cypress/e2e/users_spec/UsersCreation.cy.ts (hunk 23 above)
Comment (AnveshNalimela) [reply cypress/e2e/users_spec/UsersCreation.cy.ts]: https://github.com/AnveshNalimela/care_fe/blob/c4569fc16cb7f7f2d29552d660134c1868df9599/cypress/pageobject/Users/UserCreation.ts#L69

File: cypress/e2e/users_spec/UsersCreation.cy.ts (hunk 24 above)
Comment (AnveshNalimela) [reply cypress/e2e/users_spec/UsersCreation.cy.ts]: cy.submitButton(update_btn);

File: cypress/pageobject/Users/UserCreation.ts [hunk 25]
```diff
@@ -1,45 +1,110 @@
...
+  }
+  typeUserPhoneNumber(phone_number: string) {
+    cy.get("#phone_number").click().type(phone_number);
+  }
+  typeUserFirstName(firstName: string) {
+    cy.get("#first_name").click().type(firstName);
+  }
+  typeUserLastName(lastName: string) {
+    cy.get("#last_name").click().type(lastName);
+  }
```
[the remark this answers is not in the document]
Comment (AnveshNalimela) [reply cypress/pageobject/Users/UserCreation.ts]: one function types into filed in new user creation and other function type into differnt field

File: cypress/pageobject/Users/UserCreation.ts [hunk 26]
```diff
@@ -1,69 +1,118 @@
...
+    cy.get("#doctor_experience_commenced_on").click().type(experince);
+  }
+  typeDoctorMedicalCouncilRegNo(reg_no: string) {
+    cy.get("#doctor_medical_council_registration").click().type(reg_no);
+  }
+  typeWeeklyWorkingHours(working_hrs: string) {
+    cy.get("#weekly_working_hours").click().type(working_hrs);
+  }
+  typeAltPhoneumber(emergency_phone_number: string) {
+    cy.get("#altPhoneNumber").click().type(emergency_phone_number);
```
[the remark this answers is not in the document]
Comment (AnveshNalimela) [reply cypress/pageobject/Users/UserCreation.ts]: Ignore as they are just input paramters

**Comment by AnveshNalimela** [comment]
@nihal The deploy failed due to cache issue of apps folder.the build got failed due to import statements. Could on the issue

**Comment by deepthisharanya1615** [comment]
@rithviknishad could you check netlfiy deploy failed issue in these PR

**Comment by nihal467** [comment]
@AnveshNalimela as mentioned by the AI review, you are using mixed casing, fix it by using camelCase

File: cypress/pageobject/Users/UserCreation.ts [hunk 27]
```diff
@@ -1,45 +1,116 @@
...
-  verifyNotification(message: string) {
-    cy.verifyNotification(message);
+  selectUserType(role: string) {
+    cy.clickAndSelectOption("#user_type", role);
   }
 
+  selectHomeFacility(name: string) {
+    cy.get("#home_facility").click();
+    this.selectOptionContainingText(name);
+  }
```
[the remark this answers is not in the document]
Comment (AnveshNalimela) [reply cypress/pageobject/Users/UserCreation.ts]: Made the changes suggested

**Comment by AnveshNalimela** [comment]
@nihal467  Made the all changes could you test and merge .

**Comment by AnveshNalimela** [comment]
@nihal467  there is test case in userManage.cy.ts "linking and unlinking facility for multiple users, and confirm reflection in user cards and doctor connect" It get failed when i was trying in my local found that whenever we test for first time it get passed . But when we try for second time it fails bcoz the previous test already links faciltity to procvided user so we cant link it again. if we manually delete the linked facilties and run again . They are passed .

**Review by nihal467 — CHANGES_REQUESTED** [review]
revert unwanted changes

File: cypress/pageobject/Users/UserCreation.ts [hunk 28]
```diff
@@ -1,45 +1,122 @@
...
+    cy.clickAndSelectOption("#gender", gender);
+  }
+  selectState(state: string) {
+    cy.clickAndSelectOption("#state", state);
+  }
+  selectDistrict(district: string) {
+    cy.clickAndSelectOption("#district", district);
+  }
   selectFacility(name: string) {
     this.typeIntoInputByName("facilities", name);
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts]: - use the reusable function available in commands.ts file, to select facilities from dropdown

File: cypress/pageobject/Users/UserCreation.ts [hunk 29]
```diff
@@ -1,45 +1,122 @@
...
+    cy.get("#profile-button").click();
+  }
+  clickEditProfileButton() {
+    cy.get("#edit-cancel-profile-button").click();
+  }
+  clickAddUserButton() {
+    cy.get("#addUserButton").click();
+  }
+  clickSubmitButton() {
+    cy.get("#submit").click();
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts]: - replace them with reusable command submitButton()

File: cypress/pageobject/Users/UserCreation.ts [hunk 30]
```diff
@@ -1,45 +1,122 @@
...
+    cy.get("#user-profile-name").click();
+  }
+  clickProfileButton() {
+    cy.get("#profile-button").click();
+  }
+  clickEditProfileButton() {
+    cy.get("#edit-cancel-profile-button").click();
+  }
+  clickAddUserButton() {
+    cy.get("#addUserButton").click();
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts]: - reuse the cy.verifyAndClickElement(); to verify the button text and click it

File: cypress/pageobject/Users/UserCreation.ts [hunk 31]
```diff
@@ -1,45 +1,122 @@
 // UserCreation.ts
 export class UserCreationPage {
-  clickElementById(elementId: string) {
-    cy.get("#" + elementId).click();
+  private selectOptionContainingText(text: string) {
+    cy.get("[role='option']")
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts]: make use of the proper reusable function from command.ts for dropdown selection

File: cypress/pageobject/Users/UserCreation.ts [hunk 32]
```diff
@@ -1,45 +1,122 @@
...
+
+  typeWeeklyWorkingHours(workingHrs: string) {
+    cy.get("#weekly_working_hours").click().type(workingHrs);
+  }
+
+  typeAltPhoneNumber(emergencyPhoneNumber: string) {
+    cy.get("#altPhoneNumber").click().type(emergencyPhoneNumber);
   }
 
   typeIntoInputByName(inputName: string, value: string) {
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts]:
```suggestion

```

- no need of this, as you will be using reusable components for facility name selection

File: cypress/pageobject/Users/UserCreation.ts [hunk 33]
```diff
@@ -48,20 +125,6 @@ export class UserCreationPage {
...
-  setInputDate(dateElementId: string, dateValue: string) {
-    cy.clickAndTypeDate("#" + dateElementId, dateValue);
-  }
-
-  selectDropdownOption(dropdownId: string, optionText: string) {
-    this.clickElementById(dropdownId);
-    this.selectOptionContainingText(optionText);
-  }
-
   verifyElementContainsText(elementId: string, expectedText: string) {
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts]: - remove this type of usage and replace it with the reusable function cy.verifyContentPresence()

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 34]
```diff
@@ -53,6 +53,27 @@ describe("User Creation", () => {
...
+  const userName = "devdistrictadmin";
+  const firstName = "District Editted";
+  const lastName = "Cypress";
+  const gender = "Male";
+  const email = "test@test.com";
+  const password = "Test@123";
+  const qualification = "MBBS";
+  const experince = "2";
+  const reg_no = "123456789";
+  const firstName_ = "cypress test";
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
  const firstName = "cypress test";
```

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 35]
```diff
@@ -53,6 +53,27 @@ describe("User Creation", () => {
...
+  const lastName_ = "staff user";
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]:
```suggestion
  const lastName = "staff user";
```

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 36]
```diff
@@ -53,6 +53,27 @@ describe("User Creation", () => {
...
+  const state = "Kerala";
+  const district = "Ernakulam";
+  const role = "Doctor";
+  const home_facility = "Dummy Shifting Center";
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]: - for const naming use Camelcase

**Review by nihal467 — CHANGES_REQUESTED** [review]

File: cypress/pageobject/Users/UserCreation.ts (hunk 33 above)
Comment (AnveshNalimela) [reply cypress/pageobject/Users/UserCreation.ts]: @nihal467 These  function is called from managepatients.cy.ts so i kept it

File: cypress/pageobject/Users/UserCreation.ts [hunk 37]
```diff
@@ -1,65 +1,106 @@
 // UserCreation.ts
 export class UserCreationPage {
-  clickElementById(elementId: string) {
-    cy.get("#" + elementId).click();
+  clickProfileName() {
+    cy.get("#user-profile-name").click();
   }
```
[the remark this answers is not in the document]
Comment (AnveshNalimela) [reply cypress/pageobject/Users/UserCreation.ts]: Profile name is dyanmic we cant we verify By simply passing Profile name

File: package-lock.json [hunk 38]
```diff
@@ -115,7 +115,6 @@
     "apps/care_livekit_fe": {
       "name": "care-livekit",
       "version": "0.0.1",
-      "extraneous": true,
```
Comment (nihal467) [inline package-lock.json]: Revert the entire package-lock.json file , as its a unwanted change

File: cypress/pageobject/Users/UserCreation.ts [hunk 39]
```diff
@@ -1,65 +1,106 @@
...
+  }
+  selectDistrict(district: string) {
+    cy.clickAndSelectOption("#district", district);
   }
 
-  selectDropdownOption(dropdownId: string, optionText: string) {
-    this.clickElementById(dropdownId);
-    this.selectOptionContainingText(optionText);
+  selectFacility(name: string) {
+    cy.typeAndSelectOption("input[name='facilities']", name);
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts:45]:
```suggestion
    cy.typeAndSelectOption("input[name='facilities']", name);
```

- use the reusable component typeAndMultiSelectOption
- as mentioned in the issue, read the command.ts file to understand the existing components

File: cypress/pageobject/Users/UserCreation.ts [hunk 40]
```diff
@@ -1,65 +1,106 @@
...
+    cy.verifyAndClickElement("#profile-button", "Profile");
   }
-
-  typeIntoElementByIdPostClear(elementId: string, value: string) {
-    cy.get("#" + elementId)
-      .click()
-      .clear()
-      .click()
-      .type(value);
+  clickEditProfileButton() {
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts]:
```suggestion
  clickEditProfileButton() {
```

- we already have a clickEditProfileButton() function , dont duplicate the function, use the one in the userProfilePage.ts

File: cypress/pageobject/Users/UserCreation.ts [hunk 41]
```diff
@@ -1,65 +1,106 @@
...
-    cy.get("#" + elementId).click();
+  clickProfileName() {
+    cy.get("#user-profile-name").click();
   }
-
-  typeIntoElementById(elementId: string, value: string) {
-    cy.get("#" + elementId)
-      .click()
-      .type(value);
+  clickProfileButton() {
```
Comment (nihal467) [inline cypress/pageobject/Users/UserCreation.ts]:
```suggestion
  clickProfileButton() {
```

- reusage the navigateToProfile() in the ManageUserPage.ts file, recheck for existing function in the page object file under the Users Folder

File: cypress/e2e/users_spec/UsersCreation.cy.ts [hunk 42]
```diff
@@ -66,123 +87,90 @@ describe("User Creation", () => {
...
+    userCreationPage.typeQualification(qualification);
+    userCreationPage.typeDoctorExperience(experince);
+    userCreationPage.typeDoctorMedicalCouncilRegNo(regNo);
+    userCreationPage.typeNewUserFirstName(newUserFirstName);
+    userCreationPage.typeNewUserLastName(newUserLastName);
+    userCreationPage.typeEmail(email);
+    userCreationPage.selectGender(gender);
+    userCreationPage.selectState(state);
+    userCreationPage.selectDistrict(district);
+    cy.submitButton(saveBtn);
```
Comment (nihal467) [inline cypress/e2e/users_spec/UsersCreation.cy.ts]: - on all form submission, make sure you intercept the API with cy.intercept() and verify they are giving 200

**Review by nihal467 — CHANGES_REQUESTED** [review]
-

**Comment by nihal467** [comment]
@AnveshNalimela push the changes requested

**Comment by nihal467** [comment]
Any update

**Comment by AnveshNalimela** [comment]
@nihal i will push the requested changes tommorrow

File: cypress/e2e/users_spec/UsersCreation.cy.ts (hunk 42 above)
Comment (AnveshNalimela) [reply cypress/e2e/users_spec/UsersCreation.cy.ts]: Made the intercept for all form submission which includes the API call

File: cypress/pageobject/Users/UserCreation.ts (hunk 41 above)
Comment (AnveshNalimela) [reply cypress/pageobject/Users/UserCreation.ts]: Used the navigateToProfile()

File: cypress/pageobject/Users/UserCreation.ts (hunk 40 above)
Comment (AnveshNalimela) [reply cypress/pageobject/Users/UserCreation.ts]: Used

**Comment by AnveshNalimela** [comment]
@nihal467 Made the requested changes check out once

File: .husky/pre-commit
Comment (rithviknishad) [inline .husky/pre-commit:1]: Ensure line endings are not changed especially for files unrelated to this PR especially.

#9223 has added support for ensuring proper line endings when performing git commits and checkouts. You can reset your local index with proper line endings by merging with latest develop branch and running:

```
git rm --cached -r .
git reset --hard
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: .husky/pre-commit
Comment (AnveshNalimela) [reply .husky/pre-commit:1]: Not going to change line endings

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@CodeRabbitAI full review

**Comment by nihal467** [comment]
@coderabbitai full review

**Review by nihal467 — APPROVED** [review]

**Pull request merged**